source/includes/inkling-reference/_schema.html.md

```diff
@@ -108,102 +120,149 @@ schema MNIST_training_data_schema
+  Int64  { 0:100 }     y    # start:stop, step= 1, 0..100
+end
+```
+
+* **Colon range expression**
+
+Colon range expressions specify values for the start, 
+the step, and the stop. 
+
+* If the step value is missing, 1 is the default. 
+* Step can be  a floating point number.
+* The step size can be negative only if stop < start.
 
 ```inkling
-schema MyOutput
-  UInt8  {0,1,2,3,4}   label,    # a list of UInt8 values
-  String {"a", "bc"}   category, # a list of Strings
-  Int64  { 0:5:100 }   x,        # start:step:stop, step= 5,0..100
-  Int64  { 0:100 }     y,        # start:stop, step= 1, 0..100
-  Int64  { 0..100:25 } z,        # start:stop, numsteps=25, step= 4, 0..100
-  Float32 { 0..2:5}    a         # gives (0, .5., 1.0, 1.5, 2.0)
+# Example: Range expression, dot range type
+schema Schema4
+  Int64  { 0..100:25 } z,   # start:stop, numsteps=25
+  Float32 { 0..2:5}    a    # yields (0, .5., 1.0, 1.5, 2.0)
 end
 ```
-‍
 
-### Constrained Type Rules
+* **Dot range expression**
 
-For numeric ranges:
+Dot range expressions specify values for start, stop, and number of steps. 
 
-* For colon range, step can be  a floating point number.
-* For colon range, the step size can be negative only if stop < start.
-* For dot range, number of steps (numSteps) is a positive integer.
+* The number of steps (numSteps) must be a positive integer.
 
-### Constrained Type Syntax
+* **Numeric Range Expression Start Point**
 
-> constrainedType :=
+The start point of a numeric range is inclusive (it is included in the values of the range) and fixed. 
+The range start point is exact (to the maximum extent possible if the range expression type is floating point). 
 
-```c
-numericType
-'{'
-  start ':' [ step':']? stop // 1:2:10.   Called a 'colon range'. Specifies 'step' (default=1).
-  |
-  start '.' '.' stop ':' numSteps //   1..10:5  Called a 'dot range'. Specifies 'numsteps'.
-'}'
-```
+* **Numeric Range Expression End Point**
 
-> numericType :=
+```inkling
+ # Valid and Invalid Range Expressions.
 
-```c
-Double | Float64 | Float32 | Int8 | Int16 | Int32 |  Int64 | UInt8 | UInt16 | UInt32  | UInt64
+ Int64  { 0:4:1 }  field1,  # INVALID. step size > range.
+ Int64  { 0..1:4 } field2,  # INVALID. Values are floating point not integer.
+ Float32{-1..1:10 } field3, # VALID.   Negative bounds allowed.
+ Int8 {0:-4:-100} field4,   # VALID.   stop  < start is valid if step is negative.
+ UInt32 {-10:10}  field4    # INVALID. Unsigned integer range contains signed values.
 ```
 
+The end point specified in the range expression may or may not be included in the values of the range. 
+The range end is a limit. That means that if applying the step results in landing exactly on the end point, then the end point is part of the range. Otherwise the highest value landed on that is less than the end point is the final value in the range.
+
+Select the Inkling tab to view some examples of valid and invalid ranges.
+
 ‍
 
 [1]: #lesson
+[3]: #inkling_types
```

Comment:
/references/inkling-reference.html?#inkling-types is the link you want to use (not the new one you created), so you can short hand this to #inkling-types